shell/components/RcDropdown.vue

```diff
@@ -0,0 +1,131 @@
+<script setup lang="ts">
```

Comment:
getting in here early, just to avoid later confusion. 

the name of the component might need to be more specific to differentiate it from other drop downs (at first i thought this was a replacement for LabeledSelect)

Reply:
We had an out of band discussion on naming where we proposed keeping the name and moving the components to `pkg/rancher-components` to help with providing more context.